extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -243,7 +255,25 @@ export class AmpAd3PImpl extends AMP.BaseElement {
           this.element, undefined, opt_context);
       this.xOriginIframeHandler_ = new AmpAdXOriginIframeHandler(
           this);
-      return this.xOriginIframeHandler_.init(iframe);
+      const frameLoadPromise = this.xOriginIframeHandler_.init(iframe);
+
+      if ((this.adType == 'adsense' ||
```

Comment:
You're checking adsense or doubleclick type but only doubleclick control.  Assuming we want both